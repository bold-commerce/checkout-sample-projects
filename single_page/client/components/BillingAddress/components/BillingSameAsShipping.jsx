import { RadioField } from '@boldcommerce/stacks-ui';
import React from 'react';
import { useTranslation } from 'react-i18next';

const BillingSameAsShipping = ({
  billingSameAsShipping,
  setBillingSameAsShipping,
  disabled,
}) => {
  const { t } = useTranslation();
  return (
  <>
    <RadioField
      label={t('billing.same_as_shipping')}
      name="billing-address"
      className="RadioButton"
      value="SAME_AS_SHIPPING_ADDRESS"
      checked={billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(true)}
      disabled={disabled}
    />
    <RadioField
      label={t('billing.different_from_shipping')}
      name="billing-address"
      className="RadioButton"
      value="DIFFERENT_BILLING_ADDRESS"
      checked={!billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(false)}
      disabled={disabled}
    />
  </>
  );
};

export default BillingSameAsShipping;
