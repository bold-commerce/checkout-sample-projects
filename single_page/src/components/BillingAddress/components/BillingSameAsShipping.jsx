import { RadioField } from '@boldcommerce/stacks-ui';
import React from 'react';

const BillingSameAsShipping = ({
  billingSameAsShipping,
  setBillingSameAsShipping,
  disabled,
}) => (
  <>
    <RadioField
      label="Same as shipping address"
      name="billing-address"
      className="RadioButton"
      value="SAME_AS_SHIPPING_ADDRESS"
      checked={billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(true)}
      disabled={disabled}
    />
    <RadioField
      label="Use a different billing address"
      name="billing-address"
      className="RadioButton"
      value="DIFFERENT_BILLING_ADDRESS"
      checked={!billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(false)}
      disabled={disabled}
    />
  </>
);

export default BillingSameAsShipping;
