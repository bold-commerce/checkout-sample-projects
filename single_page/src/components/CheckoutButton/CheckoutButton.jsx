import { useBillingAddress, useCheckoutStore, useLineItems, usePaymentIframe } from '@boldcommerce/checkout-react-components';
import {
  useNavigate,
} from "react-router-dom";
import { Button } from '@boldcommerce/stacks-ui';
import React, { memo, useCallback, useState } from 'react';
import { useAnalytics, useErrorLogging } from '../../hooks';
import { useTranslation } from 'react-i18next';

const CheckoutButton = () => {
  const { processPaymentIframe } = usePaymentIframe();
  const { state } = useCheckoutStore();
  const { data: lineItems } = useLineItems();
  const { data: billingAddress } = useBillingAddress();
  const { orderStatus } = state.orderInfo;
  const processing = orderStatus === 'processing' || orderStatus === 'authorizing';

  return (
    <MemoizedCheckoutButton
      processOrder={processPaymentIframe}
      processing={processing}
      hasBillingAddress={!!billingAddress?.country_code}
      lineItems={lineItems}
      appLoading={state.loadingStatus.isLoading}
    />
  );
};

const MemoizedCheckoutButton = memo(({
  processOrder,
  processing,
  hasBillingAddress,
  lineItems,
  appLoading,
}) => {
  const trackEvent = useAnalytics();
  const logError = useErrorLogging();
  const navigate = useNavigate();
  const [loading, setLoading] = useState(false);
  const { t } = useTranslation();

  const validateInventory = useCallback(async () => {
    const variants = lineItems.map((lineItem) => lineItem.product_data.variant_id).join(',');
    const response = await fetch(`${process.env.INVENTORY_URL}?variants=${variants}`);
    const responseData = await response.json();
    let inventory = Array.from(responseData).reduce((acc,curr)=> (acc[curr.platform_id]=curr.inventory_quantity,acc),{});
    let inventoryIssues = false;

    lineItems.forEach((lineItem) => {
      inventory[lineItem.product_data.variant_id] -= lineItem.product_data.quantity;
      if (inventory[lineItem.product_data.variant_id] < 0) {
        inventoryIssues = true;
      }
    });

    if (inventoryIssues) {
      return responseData.inventory;
    } else {
      return null;
    }
  }, [lineItems]);

  const handleProcessPayment = useCallback(async () => {
    trackEvent('click_complete_order');
    setLoading(true);
    try {
      const inventoryIssues = await validateInventory();
      if (inventoryIssues) {
        setLoading(false);
        navigate('/inventory_issues', { state: inventoryIssues });
      } else {
        setLoading(false);
        await processOrder();
      }
    } catch(e) {
      logError('process_order', e);
      setLoading(false);
    }
  }, [logError, trackEvent, processOrder, validateInventory]);

  return (
    <Button
      type="button"
      className="Checkout__ConfirmButton"
      onClick={handleProcessPayment}
      loading={loading || processing}
      disabled={!hasBillingAddress || processing || appLoading}
    >
      {t('complete_order')}
    </Button>
  );
});

export default CheckoutButton;
