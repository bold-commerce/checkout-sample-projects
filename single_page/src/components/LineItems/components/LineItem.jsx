import React from 'react';
import PropTypes from 'prop-types';
import Product from './Product';
import { Price } from '@boldcommerce/stacks-ui';
import { useTranslation } from 'react-i18next';

const LineItem = ({
  title,
  image,
  quantity,
  totalPrice,
  variants
}) => {
  const { t } = useTranslation();
  return(
    <div className="SummaryBlock CartItem" >
      <Product
        title={title}
        image={image}
        variants={variants}
      />
      <div className="CartItem__QuantityPriceWrapper">
        <div className="CartItem__ProductQuantityWrapper">
          <div className="ProductQuantity" aria-label="product quantity">
            {quantity}
          </div>
        </div>
        <div className="CartItem__ProductPrice">
          <Price amount={totalPrice} moneyFormatString={t('currency_format')} />
        </div>
      </div>
    </div>
  );}
  
  LineItem.propTypes = {
    title: PropTypes.string,
    image: PropTypes.string,
    quantity: PropTypes.number,
    lineItemKey: PropTypes.string,
    variants: PropTypes.array,
  };
  
  export default LineItem;