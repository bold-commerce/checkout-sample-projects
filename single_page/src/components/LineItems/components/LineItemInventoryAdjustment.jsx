import React from 'react';
import { LineItemQuantity } from '.';
import ArrowRightIcon from './ArrowRightIcon';
import { useTranslation } from 'react-i18next';

const LineItemInventoryAdjustment = ({ originalQuantity, quantity, readOnly }) => {
  const { t } = useTranslation();
  if (quantity === 0) {
    return (
      <div className="LineItem__QuantityAdjustmentWrapper">
        <span className="LineItem__SoldOut">{t('line_item.sold_out')}</span>
      </div>
    );
  } else {
    return (
      <div className="LineItem__QuantityAdjustmentWrapper">
        <LineItemQuantity
          readOnly={readOnly}
          defaultValue={originalQuantity}
        />
        <ArrowRightIcon />
        <LineItemQuantity
          readOnly={readOnly}
          defaultValue={quantity}
        />
      </div>
    )
  }
};

export default LineItemInventoryAdjustment;
