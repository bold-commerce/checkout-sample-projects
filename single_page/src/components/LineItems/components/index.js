export { default as LineItem } from './LineItem';
export { default as Product } from './Product';
