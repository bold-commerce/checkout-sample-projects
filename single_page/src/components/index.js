export * from './Address';
export * from './BillingAddress';
export * from './CheckoutButton';
export * from './CheckoutSection';
export * from './Customer';
export * from './DiscountForm';
export * from './EmptyState';
export * from './Header';
export * from './LineItems';
export * from './LoadingState';
export * from './OrderErrors';
export * from './OrderSummary';
export * from './PaymentMethod';
export * from './ShippingAddress';
export * from './ShippingLines';
