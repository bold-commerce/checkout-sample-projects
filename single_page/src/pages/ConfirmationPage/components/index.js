export * from './ConfirmationList';
export * from './ConfirmationListItem';
