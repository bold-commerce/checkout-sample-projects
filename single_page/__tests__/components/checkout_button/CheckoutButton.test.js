import React from "react";
import { render } from "@testing-library/react";
import { CheckoutButton } from '../../../client/components/CheckoutButton';
import { exampleLineItems as MOCKlineItems } from '../../utils/lineItemHelpers';
import {
  countries as MOCKcountries,
  caProvinces as MOCKprovinces,
  exampleAddress as MOCKaddress
} from '../../utils/addressHelpers';
import { MemoryRouter } from "react-router-dom";

jest.mock('@boldcommerce/checkout-react-components', () => ({
  useBillingAddress: () => ({
    data: {MOCKaddress},
    submitBillingAddress: (() => {})
  }),
  useBillingSameAsShipping: () => ({
    setBillingSameAsShipping: (() => {}),
    data: true
  }),
  useCountryInfo: () => ({
    data: {
      countries: MOCKcountries,
      provinces: MOCKprovinces,
      provinceLabel: "province",
      showPostalCode: true,
      showProvince: true
    }
  }),
  usePaymentIframe : () => ({
    processPaymentIframe: (() => {})
  }),
  useCheckoutStore: () => ({
    state: {
      orderInfo: 'fulfilled',
      loadingStatus: {
        isLoading: false
      }
    }
  }),
  useLineItems: () => ({ 
    data : MOCKlineItems
  })
}))

describe('CheckoutButton', () => {
  test('renders CheckoutButton component', () => {
    const { asFragment } = render(
      <CheckoutButton />,
      {wrapper: MemoryRouter}
    )

    expect(asFragment()).toMatchSnapshot();
  })
})
