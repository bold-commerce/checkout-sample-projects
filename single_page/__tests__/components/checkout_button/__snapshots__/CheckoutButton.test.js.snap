// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutButton renders CheckoutButton component 1`] = `
<DocumentFragment>
  <button
    aria-busy="false"
    aria-disabled="true"
    class="stx-button stx-button--is-disabled Checkout__ConfirmButton"
    disabled=""
    type="button"
  >
    <span
      class="stx-button__text"
    >
      Confirm order
    </span>
    <span
      class="stx-loading-spinner stx-button__spinner"
    />
  </button>
</DocumentFragment>
`;
